app/src/main/java/org/oppia/app/application/OppiaApplication.kt

```diff
@@ -30,4 +32,8 @@ class OppiaApplication :
     FirebaseApp.initializeApp(applicationContext)
     component.getApplicationStartupListeners().forEach(ApplicationStartupListener::onCreate)
   }
+
+  override fun getWorkManagerConfiguration(): Configuration {
```

Comment:
This should be done via modules not directly in components.

Reply:
Done.

Reply:
I tried implementing this via the module and here's what I did - 
Implemented the `Configuration.Provider` in the WorkManagerInitialiser and then initialised the work manager in its on create and then bound them in the module. I had to remove the default work manager initialisation using a content provider in the manifest. All of this worked perfectly in production but didn't seem to work in tests as the work manager was getting initialised again and again. 

So I changed the implementation to just providing the configuration via module and then returning that configuration in the `getWorkManagerConfiguration()`. This way production worked, tests worked and we didn't need to alter the default initialisation of the work manager. Is this fine ?

Reply:
I'm slightly confused. I actually would expect the work manager to be initialized repeatedly in tests since each test recreates the test application.

With this solution, don't we need to update all test applications to also be a Configuration.Provider? How does it actually get hooked up?

Reply:
I also expected the same but once it's initialised, it gives error when we re-initialise it again. Also, it got hooked up to the tests as is. Whenever we want to use their functionality, we can initialise the testWorkManager via the api provided by the testing dependency.

Reply:
Looks like this is resolved now.